src/NCalc/LambdaExpressionVistor.cs

```diff
@@ -200,26 +230,76 @@ public override void Visit(Function function)
                 return;
             }
 
-            switch (functionName)
+            Action<string, int, int> CheckArgumentsLengthForFunction = delegate (string funcStr, int argsNum, int argsNeed)
             {
-                case "MIN":
-                    var minArg0 = L.Expression.Convert(args[0], typeof(double));
-                    var minArg1 = L.Expression.Convert(args[1], typeof(double));
-                    _result = L.Expression.Condition(L.Expression.LessThan(minArg0, minArg1), minArg0, minArg1);
-                    break;
-                case "MAX":
-                    var maxArg0 = L.Expression.Convert(args[0], typeof(double));
-                    var maxArg1 = L.Expression.Convert(args[1], typeof(double));
-                    _result = L.Expression.Condition(L.Expression.GreaterThan(maxArg0, maxArg1), maxArg0, maxArg1);
-                    break;
-                case "POW":
-                    var powArg0 = L.Expression.Convert(args[0], typeof(double));
-                    var powArg1 = L.Expression.Convert(args[1], typeof(double));
-                    _result = L.Expression.Power(powArg0, powArg1);
-                    break;
-                default:
-                    throw new MissingMethodException($"method not found: {functionName}");
+                if (argsNum != argsNeed)
+                    throw new ArgumentException($"{funcStr} takes exactly {argsNeed} argument");
+            };
+
+            Action<string, int> MakeMathCallExpressionOneDoubleParam = delegate (string mathCallFuncName, int argsNumActual)
+            {
+                CheckArgumentsLengthForFunction(mathCallFuncName, argsNumActual, 1);
+                var arg = L.Expression.Convert(args[0], typeof(double));
+                _result = L.Expression.Call(typeof(Math).GetMethod(mathCallFuncName), arg);

```

Comment:
It seems like this would use reflection much more than necessary.  Every time a math function is encountered you're using reflection to get the method.  I think you should be able to do the lookup once via a static property and reuse it indefinitely.

Reply:
Sounds good